datafusion/common/src/config.rs

```diff
@@ -204,6 +204,12 @@ config_namespace! {
         /// MySQL, PostgreSQL, Hive, SQLite, Snowflake, Redshift, MsSQL, ClickHouse, BigQuery, and Ansi.
         pub dialect: String, default = "generic".to_string()
 
+        /// If set to true, the system will return an error when encountering a `Varchar`
+        /// type with a specified length. This can be useful for enforcing certain schema
+        /// constraints or maintaining compatibility with systems that do not support
+        /// length-specified `Varchar` types.
+        pub error_on_varchar_with_length: bool, default = false
```

Comment:
It might be easier to understand to avoid the double negative here

So perhaps the config flag could be `support_varchar_with_length` and default to `true` 🤔

Reply:
sure, I'll fix it